### ISSUE TITLE
Show breadcrumbs whilst debugging

### ISSUE THREAD

**Issue by tekumara** [microsoft/vscode#147636]
I have breadcrumbs enabled but they do not show during debugging.

You can see here that the file I'm debugging (which is a third party library) does not have breadcrumbs. Files I open directly do have breadcrumbs:

![Recording 2022-04-18 at 12 26 31](https://user-images.githubusercontent.com/125105/163744886-ea71bedc-8cb7-4027-9995-3fb00dd4a2d3.gif)

**Comment by jrieken** [comment]
> You can see here that the file I'm debugging (which is a third party library) does not have breadcrumbs.

@karthiknadig How does python "serve" these files? Is it via the `TextDocumentContentProvider`-API?

**Comment by karthiknadig** [comment]
@jrieken This happens via DAP. So the file that is opening here is because we tell that the top of the stack. We don't serve files for debugging.

/cc @int19h

**Comment by int19h** [comment]
DAP itself has a [message that client can use to request the source code](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source), and debugpy does support it. Outside of Jupyter, however, this logic should only kick in when doing remote debugging with path mapping - unmapped remote files are fetched like that. 

IIRC if you hover over the tab for this file in the editor, you should see the remote path for it. This might provide some clue as to why it wasn't path-mapped.

**Comment by jrieken** [comment]
> This happens via DAP. So the file that is opening here is because we tell that the top of the stack. We don't serve files for debugging.

@weinand Is it that the DAP implementation uses some form of virtual document for this, e.g something that isn't backed by a real or virtual file system but just a readonly text file?

**Comment by weinand** [comment]
@jrieken yes, whenever the debugger hits a breakpoint in a location not backed by a real file, the debugger sends a "debug URI" to VS Code via DAP. VS Code resolves the URI via the [`DebugContentProvider`](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/common/debugContentProvider.ts ) which loads the readonly text contents via DAP's `source` request.

**Comment by jrieken** [comment]
Thanks for clarifying.

Today breadcrumbs hides/disables whenever the document isn't from a file system. That's because its path cannot be "browsed", e.g the path-component of its uri must not be separated by the "normal" separator character and there is no way to detect siblings along the path (`readdir` cannot be called for these uris). That something breadcrumbs cannot change but the handling of this can be improved. Ideas

* show the uri path but don't allow for browsing sibling
* still show the symbol path

**Comment by gjsjohnmurray** [comment]
Possible duplicate of #62290
